src/helpers/setPath.js

```diff
@@ -30,6 +31,10 @@ function setPath(path, val, obj) {
   const key = path[0]
   let newVal = val
 
+  if(!(isString(key) && !isEmpty(key) || isInteger(key) && key >= 0)) {
```

Comment:
This appears in a few spots, do we want to put it somewhere common or we don't care?

Reply:
We will more than likely abstract this when we move `prop` -> `getProp` and `propPath` -> `getPath`.